### PR TITLE
pump spark version from 3.2.2 to 3.3.1

### PR DESCRIPTION
**Pull request by JessicaXYWang**
## Related Issues/PRs

# Testing pump spark version from 3.2.2 to 3.3.0, submit PR to get a snapshot.

## What changes are proposed in this pull request?

Pump Spark Version from 3.2.2 to 3.3.0

## How is this patch tested?

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change any dependencies?

- [x] No. You can skip this section.
- [ ] Yes. Make sure the dependencies are resolved correctly, and list changes here.

## Does this PR add a new feature? If so, have you added samples on website?

- [x] No. You can skip this section.
- [ ] Yes. Make sure you have added samples following below steps.

1. Find the corresponding markdown file for your new feature in `website/docs/documentation` folder.
   Make sure you choose the correct class `estimators/transformers` and namespace.
2. Follow the pattern in markdown file and add another section for your new API, including pyspark, scala (and .NET potentially) samples.
3. Make sure the `DocTable` points to correct API link.
4. Navigate to website folder, and run `yarn run start` to make sure the website renders correctly.
5. Don't forget to add `<!--pytest-codeblocks:cont-->` before each python code blocks to enable auto-tests for python samples.
6. Make sure the `WebsiteSamplesTests` job pass in the pipeline.

### CODE REVIEW

File: core/src/test/scala/com/microsoft/azure/synapse/ml/nbtest/DatabricksUtilities.scala [hunk 1]
```diff
@@ -31,8 +31,9 @@ object DatabricksUtilities {
   val Region = "eastus"
   val PoolName = "synapseml-build-10.4"
   val GpuPoolName = "synapseml-build-10.4-gpu"
```
Comment (mhamilton723) [inline core/src/test/scala/com/microsoft/azure/synapse/ml/nbtest/DatabricksUtilities.scala:33]: You will need to make new pools on our build server too

**Review by mhamilton723 — CHANGES_REQUESTED** [review]
I still think there might be more instances of 3.2.2 or 3.2.0 hiding in the code. use Intelli-J's ctrl-shift-r to find and replace in all files in the project. Once you are in this menu, open the results in the "find menu" and you can work your way through all 3.2.2's in an organized way

File: core/src/main/scala/com/microsoft/azure/synapse/ml/explainers/RegressionBase.scala [hunk 2]
```diff
@@ -43,6 +43,7 @@ abstract class RegressionBase {
     * when running on Spark 3.0.* and 3.1.*.
     * Workaround: use reflection to construct the implementation.
     */
+  //TODO: Check for spark 3.3.0
```
Comment (memoryz) [inline core/src/main/scala/com/microsoft/azure/synapse/ml/explainers/RegressionBase.scala:46]: Anything I can help check?

File: .github/workflows/ci-tests-r.yml [hunk 3]
```diff
@@ -41,7 +41,7 @@ jobs:
         activate-environment: synapseml
     - name: Download Spark
       shell: bash -l {0}
-      run: curl https://archive.apache.org/dist/spark/spark-3.2.0/spark-3.2.0-bin-hadoop3.2.tgz -o spark-3.2.0-bin-hadoop3.2.tgz
```
Comment (mhamilton723) [inline .github/workflows/ci-tests-r.yml:44]: you are deleting the 3.2.tgz here by mistaker

File: tools/dotnet/dotnetSetup.sh [hunk 4]
```diff
@@ -20,11 +20,11 @@ echo "##vso[task.setvariable variable=DOTNET_WORKER_DIR]$DOTNET_WORKER_DIR"
 # Install Sleet
 dotnet tool install -g sleet
 
-# Install Apache Spark-3.2
-curl https://archive.apache.org/dist/spark/spark-3.2.0/spark-3.2.0-bin-hadoop3.2.tgz -o spark-3.2.0-bin-hadoop3.2.tgz
+# Install Apache Spark-3.3
+curl https://archive.apache.org/dist/spark/spark-3.3.0/spark-3.3.0-bin-hadoop3.tgz -o spark-3.3.0-bin-hadoop3.tgz
```
Comment (mhamilton723) [inline tools/dotnet/dotnetSetup.sh:24]: error here too with deleting hadoop 3.2

**Pull request closed without merging**